ports/webassembly/library.js

```diff
@@ -33,10 +33,9 @@ mergeInto(LibraryManager.library, {
                 process.stdout.write(b);
             } else {
                 var c = String.fromCharCode(getValue(ptr + i, 'i8'));
-                var mp_js_stdout = document.getElementById('mp_js_stdout');
-                var print = new Event('print');
-                print.data = c;
-                mp_js_stdout.dispatchEvent(print);
+                var printEvent = new CustomEvent('micropython-print');
+                printEvent.data = c;
```

Comment:
`CustomEvent` already has a `detail` property that should be used instead of adding a custom property.

```suggestion
var printEvent = new CustomEvent('micropython-print', { detail: c });
```

Reply:
This was already merged.  That change will need to go in a new PR (maybe @ntoll will get to it with the above-discussed improvements).